From 71992b65d6b4623cec4f3262a954b47e61c2284a Mon Sep 17 00:00:00 2001
From: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Date: Tue, 30 May 2023 02:45:07 +0100
Subject: [PATCH 100/173] arm64: dts: qcom: sc8280xp-x13s: Define the RGB
 sensor on cci2_i2c1

The ov5675 is at i2c address 0x10 read 0x20/write 0x21 and is located on pins
cci_i2c_sda5/cci_i2c_scl5. MCLK3 is used to clock the chip with vreg_l7q,
vreg_l6q and vreg_l2q providing dovdd, avdd and dvdd respectively.

Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
---
 .../qcom/sc8280xp-lenovo-thinkpad-x13s.dts    | 67 +++++++++++++++++++
 1 file changed, 67 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
index 22d4dd410dfc..4816e2034ad6 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
@@ -575,6 +575,73 @@ &camcc {
 	status = "okay";
 };
 
+&camss {
+	/* CSIPHY0 CSIPHY1 1.2v = l4d, 0.88v = l6d
+	 * TODO: CSIPHY2, CSIPHY3 1.2v = l3b, 0.88v = l6b
+	 */
+	vdda-phy-supply = <&vreg_l6d>;
+	vdda-pll-supply = <&vreg_l4d>;
+
+	status = "okay";
+
+	ports {
+		/* The port index denotes CSIPHY id i.e. csiphy0 */
+		port@0 {
+			csiphy0_lanes01_ep: endpoint@0 {
+				reg = <0>;
+				clock-lanes = <7>;
+				data-lanes = <0 1>;
+				remote-endpoint = <&ov5675_ep>;
+			};
+		};
+	};
+};
+
+&cci2 {
+	status = "okay";
+};
+
+&cci2_i2c1 {
+	/*
+	 * CCI_I2C5_SDA
+	 * CCI_I2C5_SCL
+	 * CAM_MCLK3
+	 * CAMERA_A0_CLK_M
+	 * CAMERA_NC_CLK_P
+	 * CAMERA_B1_LN1_M
+	 * CAMERA_A1_LN1_P
+	 * CAMERA_C0_LN0_M
+	 * CAMERA_B0_LN0_P
+	 */
+	camera@10 {
+		compatible = "ovti,ov5675";
+		reg = <0x10>;
+
+		reset-gpios = <&tlmm 15 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cam_rgb_default>;
+
+		clocks = <&camcc CAMCC_MCLK3_CLK>;
+		assigned-clock-rates = <19200000>;
+
+		orientation = <0>;	/* Front facing */
+
+		avdd-supply = <&pm8008_l6>;
+		dvdd-supply = <&pm8008_l2>;
+		dovdd-supply = <&pm8008_l7>;
+
+		port {
+			ov5675_ep: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				link-frequencies = /bits/ 64 <450000000>;
+				remote-endpoint = <&csiphy0_lanes01_ep>;
+			};
+		};
+
+	};
+};
+
 &dispcc0 {
 	status = "okay";
 };
-- 
2.43.0

